chain/network/src/peer_manager.rs

```diff
@@ -435,10 +725,17 @@ impl PeerManagerActor {
         // update that represents the connection removal.
         self.active_peers.remove(&peer_id);
 
+        #[cfg(feature = "protocol_feature_routing_exchange_algorithm")]
+        self.ibf_contructor_pool
+            .send(IbfSetConstructorMessages::RemovePeer(peer_id.clone()))
```

Comment:
This seems to indicate that we need `RemovePeer` outside of adversarial settings.

Reply:
Yes, we do this on disconnect, which is intentional.